### PR TITLE
contract clean up for distribution

### PR DESCRIPTION
**Pull request by ghostffcode**

### CODE REVIEW

File: packages/hardhat/contracts/Distributor.sol [hunk 1]
```diff
@@ -26,49 +26,92 @@
...
+        for (uint256 i = 0; i < recipients.length; i++) {
+            require(
+                recipients[i] != address(0),
+                "You can't distribute to the genesis address"
+            );
+            require(values[i] > 0, "Why waste gas for zero Ether send");
+            total += values[i];
+        }
+
+        require(msg.value >= total, "Not enough Ether to distribute");
```
Comment (hansmrtn) [inline packages/hardhat/contracts/Distributor.sol:66]: Could just be a hardhat node thing, but this requirement breaks the current tests in the latest tests on the dev branch

File: packages/hardhat/contracts/Distributor.sol [hunk 2]
```diff
@@ -26,49 +26,92 @@
...
+        }
+
+        emit ethDistributed(msg.sender, name);
     }
-    require(token.transferFrom(msg.sender, address(this), total));
-    for (uint256 i = 0; i < recipients.length; i++) {
-      require(token.transfer(recipients[i], values[i]));
+
+    function distributeToken(
+        IERC20 token,
```
Comment (hansmrtn) [inline packages/hardhat/contracts/Distributor.sol:82]: ERC20 implements the internal method ` _transfer(address sender, address recipient, uint256 amount)` with requirements about spenders balances, null addresses, and allowances. Does that make these `require`ments redundant?

**Review by hansmrtn — COMMENTED** [review]
Minus the the tests that need to be written, this looks pretty solid! Check out the comments/questions about requirements. 

Also worth noting, this contract is about 2X the gas to deploy.

**Review by hansmrtn — APPROVED** [review]
It makes sense to not rely on the implementation of ERC20 contracts and instead have our own checks. In fact, we should look into using raricaptitals/solmates erc20 implementation

**Review by farque65 — APPROVED** [review]

**Pull request merged**